CHANGELOG.md

```diff
@@ -3,592 +3,117 @@
-- Skip unquote allocation for non-quoted strings ([#11813](https://github.com/astral-sh/uv/pull/11813))
-- Use `SmallString` for filenames and URLs ([#11765](https://github.com/astral-sh/uv/pull/11765))
-- Use a Boxed slice for version specifiers ([#11766](https://github.com/astral-sh/uv/pull/11766))
-- Use matches over contains for extra value parsing ([#11770](https://github.com/astral-sh/uv/pull/11770))
+  uv 0.7.0 (4433f41c9 2025-04-29)
+  ```
 
-### Documentation
+  As described in the warning, `--preview` can be used to error instead:
 
-- Avoid fallback to PyPI in mixed CPU/CUDA example ([#11115](https://github.com/astral-sh/uv/pull/11115))
-- Docs: Clarify that setting cache-keys overrides defaults ([#11895](https://github.com/astral-sh/uv/pull/11895))
-- Document our MSRV policy ([#11898](https://github.com/astral-sh/uv/pull/11898))
-- Fix reference to macOS cache path ([#11845](https://github.com/astral-sh/uv/pull/11845))
-- Fix typo in `no_default_groups` documentation and changelog ([#11928](https://github.com/astral-sh/uv/pull/11928))
-- Update the "Locking and syncing" page ([#11647](https://github.com/astral-sh/uv/pull/11647))
-- Update alternative indexes documentation to use new interface ([#10826](https://github.com/astral-sh/uv/pull/10826))
+  ```console
+  $ uv version --preview
+  error: No `pyproject.toml` found in current directory or any parent directory
+  ```
 
-## 0.6.3
+  The previous functionality of `uv version` was moved to `uv self version`.
 
-### Enhancements
+- **Avoid fallback to subsequent indexes on authentication failure ([#12805](https://github.com/astral-sh/uv/pull/12805))**
 
-- Allow quotes around command-line options in `requirement.txt files` ([#11644](https://github.com/astral-sh/uv/pull/11644))
-- Initialize PEP 723 script in `uv lock --script` ([#11717](https://github.com/astral-sh/uv/pull/11717))
+  When using the `first-index` strategy (the default), uv will stop searching indexes for a package once it is found on a single index. Previously, uv considered a package as "missing" from an index during authentication failures, such as an HTTP 401 or HTTP 403 (normally, missing packages are represented by an HTTP 404). This behavior was motivated by unusual responses from some package indexes, but reduces the safety of uv's index strategy when authentication fails. Now, uv will consider an authentication failure as a stop-point when searching for a package across indexes. The `index.ignore-error-codes` option can be used to recover the existing behavior, e.g.:
 
-### Configuration
+  ```toml
+  [[tool.uv.index]]
+  name = "pytorch"
+  url = "https://download.pytorch.org/whl/cpu"
+  ignore-error-codes = [401, 403]
+  ```
 
-- Accept multiple `.env` files in `UV_ENV_FILE` ([#11665](https://github.com/astral-sh/uv/pull/11665))
+  Since `pytorch`'s indexes always return a HTTP 403 for missing packages, uv special-cases indexes on the `pytorch.org` domain to ignore that error code by default.
```

Comment:
```suggestion
  Since PyTorch's indexes always return a HTTP 403 for missing packages, uv special-cases indexes on the `pytorch.org` domain to ignore that error code by default.
```